clients/fides-js/src/components/Overlay.tsx

```diff
@@ -46,26 +55,41 @@ const Overlay: FunctionComponent<Props> = ({
   const hasMounted = useHasMounted();
   const [bannerIsOpen, setBannerIsOpen] = useState(false);
 
+  const dispatchCloseEvent = () => {
+    dispatchFidesEvent(
+      "FidesModalClosed",
+      getLatestCookie(cookie),
+      options.debug
+    );
+  };
+
   const { instance, attributes } = useA11yDialog({
     id: "fides-modal",
     role: "dialog",
     title: experience?.experience_config?.title || "",
+    onClose: dispatchCloseEvent,
   });
 
   const handleOpenModal = useCallback(() => {
     if (instance) {
       instance.show();
-      dispatchFidesEvent("FidesUIShown", cookie, options.debug, {
-        servingComponent: ServingComponent.OVERLAY,
-      });
+      dispatchFidesEvent(
+        "FidesUIShown",
+        getLatestCookie(cookie),
+        options.debug,
+        {
+          servingComponent: ServingComponent.OVERLAY,
+        }
+      );
     }
   }, [instance, cookie, options.debug]);
 
   const handleCloseModal = useCallback(() => {
     if (instance) {
       instance.hide();
+      dispatchCloseEvent();
     }
-  }, [instance]);
+  }, [instance, cookie, options.debug]);
```

Comment:
I think the dependency array here would be 
```suggestion
  }, [instance, dispatchCloseEvent]);
```